batch/batch/front_end/validate.py

```diff
@@ -66,13 +81,69 @@ def __init__(self, reason):
         self.reason = reason
 
 
-def validate_jobs(jobs):
+def validate_and_clean_jobs(jobs):
     if not isinstance(jobs, list):
         raise ValidationError('jobs is not list')
     for i, job in enumerate(jobs):
+        handle_deprecated_args(i, job)
         validate_job(i, job)
 
 
+def handle_deprecated_args(i, job):
+    if 'pvc_size' in job:
+        if 'storage' in resources:
+            raise ValidationError(f"jobs[{i}].resources.storage is already defined, but "
+                                  f"deprecated key 'pvc_size'.")
+        deprecated_msg = "[pvc_size key is DEPRECATED. Use resources.storage]"
+        pvc_size = job['pvc_size']
+        if not isinstance(pvc_size, str):
+            raise ValidationError(f'{deprecated_msg} jobs[{i}].pvc_size not str.')
```

Comment:
Double validating e.g. here (here, and subsequent validation pass on `resources.storage`) to return less confusing error messages to old clients.
Once we can send messages like "this is deprecated" to the client, I'd like to remove validation logic in the `handle_deprecated_args` pass.